plugins/directory-listing/src/main/java/com/blazemeter/jmeter/DirectoryListing.java

```diff
@@ -0,0 +1,170 @@
+package com.blazemeter.jmeter;
+
+import org.apache.jmeter.config.ConfigTestElement;
+import org.apache.jmeter.engine.event.LoopIterationEvent;
+import org.apache.jmeter.engine.event.LoopIterationListener;
+import org.apache.jmeter.threads.JMeterContextService;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+public class DirectoryListing extends ConfigTestElement implements LoopIterationListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DirectoryListing.class);
+
+    public static final String SOURCE_DIRECTORY = "directory";
+    public static final String DESTINATION_VARIABLE_NAME = "variableName";
+    public static final String USE_FULL_PATH = "useFullPath";
+    public static final String RANDOM_ORDER = "randomOrder";
+    public static final String RECURSIVE_LISTING = "recursiveListing";
+    public static final String REWIND_ON_THE_END = "rewindOnTheEndOfList";
+    public static final String INDEPENDENT_LIST_PER_THREAD = "independentListPerThread";
+    public static final String RE_READ_DIRECTORY_ON_THE_END_OF_LIST = "reReadDirectory";
+
+    private Iterator<File> iterator;
+    private List<File> list;
+
+    @Override
+    public void iterationStart(LoopIterationEvent loopIterationEvent) {
+        // TODO: independentListPerThread
+
+        try {
+            if (list == null) {
+                list = getDirectoryListing();
+                iterator = list.iterator();
+            }
+
+            if (!iterator.hasNext()) {
+                boolean isReRead = getReReadDirectoryOnTheEndOfList();
+                boolean isRewindOnTheEnd = getRewindOnTheEnd();
+
+                if (isReRead && isRewindOnTheEnd) {
+                    this.list = getDirectoryListing();
+                    this.iterator = list.iterator();
+                } else if (isRewindOnTheEnd) {
+                    this.iterator = list.iterator();
+                } else {
+                    // if the end of list && !isRewindOnTheEnd
+                    return;
+                }
+            }
+
+        } catch (FileNotFoundException ex) {
```

Comment:
yes, here we must catch this exception and logging it or throw RuntimeException (failed all test)